<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to fragquaxi • fragquaxi</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to fragquaxi">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">fragquaxi</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/fragquaxi.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to fragquaxi</h1>
            
      
      
      <div class="hidden name"><code>fragquaxi.Rmd</code></div>

    </div>

    
    
<p>The <code>fragquaxi</code> package allows you to obtain <strong>fr</strong>actional <strong>a</strong>bundances of <strong>g</strong>lycoforms (and proteoforms in general) from mass spectrometric (MS) data by <strong>qua</strong>ntification <em>via</em> <strong>X</strong>IC (extracted ion current) <strong>i</strong>ntegration.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(purrr)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(tidyr)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(mzR)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(fragquaxi)</a></code></pre></div>
<div id="molecular-formulas" class="section level1">
<h1 class="hasAnchor">
<a href="#molecular-formulas" class="anchor"></a>Molecular formulas</h1>
<p>This package provides an S3 class <code>mol_form</code> that represents molecular formulas. Such a formula is created by <code><a href="../reference/molecular_formula.html">molecular_formula()</a></code>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">emtansine &lt;-<span class="st"> </span><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C35 H48 Cl1 N3 O10 S1"</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">emtansine</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; Molecular formula &lt;C35 H48 Cl N3 O10 S&gt;</span></a></code></pre></div>
<p>Spaces separating individual elements are optional, as are counts with value 1. Hence, the following function calls will return identical formula objects:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C35 H48 Cl1 N3 O10 S1"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; Molecular formula &lt;C35 H48 Cl N3 O10 S&gt;</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C35 H48 Cl N3 O10 S"</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; Molecular formula &lt;C35 H48 Cl N3 O10 S&gt;</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C35H48ClN3O10S"</span>)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; Molecular formula &lt;C35 H48 Cl N3 O10 S&gt;</span></a></code></pre></div>
<p>Element counts may be negative or marked by an explicit plus sign:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"N-1 H-3"</span>)      <span class="co"># Gln -&gt; pyroglutamate</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt; Molecular formula &lt;H-3 N-1&gt;</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"P+1 O+3 H+1"</span>)  <span class="co"># phosphorylation</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; Molecular formula &lt;H O3 P&gt;</span></a></code></pre></div>
<p>Note how the order of elements in the printed output complies with the Hill system.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"Cl4 C"</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt; Molecular formula &lt;C Cl4&gt;</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"H12 C6 O6"</span>)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; Molecular formula &lt;C6 H12 O6&gt;</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6">  </a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"H2SO4"</span>)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; Molecular formula &lt;H2 O4 S&gt;</span></a></code></pre></div>
<p><code><a href="../reference/molecular_formula.html">molecular_formula()</a></code> is vectorized: If called with a string vector, it returns a list of formula objects.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"H2 O"</span>, <span class="st">"C6 H12 O6"</span>))</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#&gt; [[1]]</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; Molecular formula &lt;H2 O&gt;</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt; [[2]]</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt; Molecular formula &lt;C6 H12 O6&gt;</span></a></code></pre></div>
<p>Formulas support elementary arithmetic operations. (In case of multiplication, the numeric multiplier is always coerced to integer <em>via</em> <code><a href="https://rdrr.io/r/base/integer.html">as.integer()</a></code> in order to ensure that element counts remain integral).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">f1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C6 H10 O5"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">f2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C5 H11 NO2 S"</span>)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">f1 <span class="op">+</span><span class="st"> </span>f2</a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; Molecular formula &lt;C11 H21 N O7 S&gt;</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"></a>
<a class="sourceLine" id="cb7-6" data-line-number="6">f1 <span class="op">-</span><span class="st"> </span>f2</a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; Molecular formula &lt;C H-1 N-1 O3 S-1&gt;</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"></a>
<a class="sourceLine" id="cb7-9" data-line-number="9">f1 <span class="op">*</span><span class="st"> </span>2L</a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; Molecular formula &lt;C12 H20 O10&gt;</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"></a>
<a class="sourceLine" id="cb7-12" data-line-number="12">f1 <span class="op">*</span><span class="st"> </span><span class="fl">2.9</span>  <span class="co"># multiplier implicitly converted to the integer 2L</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; Molecular formula &lt;C12 H20 O10&gt;</span></a></code></pre></div>
<p><code><a href="../reference/get_mass.html">get_mass()</a></code> calculates the molecular mass of a formula:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C6 H10 O5"</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/get_mass.html">get_mass</a></span>()</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#&gt; [1] 162.141</span></a></code></pre></div>
<p>By default, <code><a href="../reference/get_mass.html">get_mass()</a></code> uses average atomic masses. However, the package also provides monoisotopic masses:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C6 H10 O5"</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/get_mass.html">get_mass</a></span>(<span class="dt">mass_set =</span> <span class="st">"monoisotopic"</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; [1] 167.0528</span></a></code></pre></div>
<p>The dataset <code>atomic_masses</code> provides these average and monoisotopic masses:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">atomic_masses</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#&gt; # A tibble: 118 x 6</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#&gt;    element name          Z average     A monoisotopic</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;     &lt;int&gt;   &lt;dbl&gt; &lt;int&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#&gt;  1 H       hydrogen      1    1.01     1         1.01</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#&gt;  2 He      helium        2    4.00     4         4.00</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt;  3 Li      lithium       3    6.94     7         7.02</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt;  4 Be      beryllium     4    9.01     9         9.01</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="co">#&gt;  5 B       boron         5   10.8     11        11.0 </span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10"><span class="co">#&gt;  6 C       carbon        6   12.0     12        12   </span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co">#&gt;  7 N       nitrogen      7   14.0     14        14.0 </span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co">#&gt;  8 O       oxygen        8   16.0     16        17.0 </span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"><span class="co">#&gt;  9 F       fluorine      9   19.0     19        20.0 </span></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="co">#&gt; 10 Ne      neon         10   20.2     20        21.0 </span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="co">#&gt; # … with 108 more rows</span></a></code></pre></div>
<p><code><a href="../reference/new_mass_set.html">new_mass_set()</a></code> allows you to define custom mass sets:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">my_masses &lt;-<span class="st"> </span><span class="kw"><a href="../reference/new_mass_set.html">new_mass_set</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">H =</span> <span class="dv">10</span>, <span class="dt">C =</span> <span class="dv">100</span>, <span class="dt">O =</span> <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"C6 H8 O5"</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/get_mass.html">get_mass</a></span>(my_masses)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt; [1] 685</span></a></code></pre></div>
<p>In order to avoid missing values in custom mass sets, you may choose to inherit masses from a previously defined set <em>via</em> the argument <code>inherits_from</code>:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">my_masses_complete &lt;-<span class="st"> </span><span class="kw"><a href="../reference/new_mass_set.html">new_mass_set</a></span>(</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">  <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">H =</span> <span class="dv">10</span>, <span class="dt">C =</span> <span class="dv">100</span>, <span class="dt">O =</span> <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">  <span class="dt">inherits_from =</span> <span class="st">"average"</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4">)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"H2 S O4"</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/get_mass.html">get_mass</a></span>(my_masses)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#&gt; [1] NA</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(<span class="st">"H2 S O4"</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="../reference/get_mass.html">get_mass</a></span>(my_masses_complete)</a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt; [1] 56.06</span></a></code></pre></div>
</div>
<div id="define-proteoforms-and-ions" class="section level1">
<h1 class="hasAnchor">
<a href="#define-proteoforms-and-ions" class="anchor"></a>Define proteoforms and ions</h1>
<p><code><a href="../reference/charge.html">charge()</a></code> converts a molecular mass to a mass-to-charge ratio.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/charge.html">charge</a></span>(<span class="dv">150000</span>, <span class="dt">z =</span> <span class="dv">24</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co">#&gt;       24 </span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="co">#&gt; 6251.008</span></a></code></pre></div>
<p>The function is vectorized over charge <code>z</code>:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="../reference/charge.html">charge</a></span>(<span class="dv">150000</span>, <span class="dt">z =</span> <span class="dv">20</span><span class="op">:</span><span class="dv">24</span>)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#&gt;       20       21       22       23       24 </span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt; 7501.008 7143.865 6819.190 6522.747 6251.008</span></a></code></pre></div>
<p><code><a href="../reference/charge.html">charge()</a></code> naturally works with masses calculated for molecular formulas:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">mab_atoms &lt;-<span class="st"> "C6464 H9950 N1706 O2014 S44"</span>  <span class="co"># formula of a monoclonal antibody</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw"><a href="../reference/molecular_formula.html">molecular_formula</a></span>(mab_atoms) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/get_mass.html">get_mass</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="../reference/charge.html">charge</a></span>(<span class="dv">20</span><span class="op">:</span><span class="dv">24</span>)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="co">#&gt;       20       21       22       23       24 </span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; 7260.872 6915.164 6600.884 6313.933 6050.894</span></a></code></pre></div>
<p>Above, the molecular formula of the antibody was determined by an external tool. However, you may also use <code><a href="../reference/load_protein_sequence.html">load_protein_sequence()</a></code> to load a protein sequence in FASTA format and obtain its formula. The FASTA file must contain one record for each protein chain (so that the formula reflects the correct number of chain termini). In addition, you must indicate the number of disulfide bridges (if any) <em>via</em> the optional argument <code>disulfides</code>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">mab_sequence &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">  <span class="st">"extdata"</span>, <span class="st">"mab_sequence.fasta"</span>,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">  <span class="dt">package =</span> <span class="st">"fragquaxi"</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4">)</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">mab_formula &lt;-<span class="st"> </span><span class="kw"><a href="../reference/load_protein_sequence.html">load_protein_sequence</a></span>(mab_sequence, <span class="dt">disulfides =</span> <span class="dv">16</span>)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6"></a>
<a class="sourceLine" id="cb16-7" data-line-number="7">mab_formula</a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="co">#&gt; Molecular formula &lt;C6464 H9950 N1706 O2014 S44&gt;</span></a></code></pre></div>
<p><code>fragquaxi</code> is tailored towards analysis of <em>proteoforms</em>, <em>i.e.</em>, the different molecular forms in which the protein product of a single gene can be found due to changes introduced by posttranslational modifications (PTMs). In the context of this package, a set of proteoforms is described by a data frame:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">proteoforms &lt;-<span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">tribble</a></span>(</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">  <span class="op">~</span>name,       <span class="op">~</span>Hex, <span class="op">~</span>HexNAc, <span class="op">~</span>Fuc, <span class="op">~</span>PHOS, <span class="op">~</span>foo,</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">  <span class="st">"G0F/G0"</span>,       <span class="dv">6</span>,       <span class="dv">8</span>,    <span class="dv">1</span>,    <span class="dv">0</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">  <span class="st">"G0F/G0F"</span>,      <span class="dv">6</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">0</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">  <span class="st">"G0F/G1F"</span>,      <span class="dv">7</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">0</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">  <span class="st">"G1F/G1F"</span>,      <span class="dv">8</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">0</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">  <span class="st">"G1F/G2F"</span>,      <span class="dv">9</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">0</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-8" data-line-number="8">  <span class="st">"G2F/G2F"</span>,     <span class="dv">10</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">0</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">  <span class="st">"G2F/G2F+P"</span>,   <span class="dv">10</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">1</span>,     <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">  <span class="st">"G2F/G2F+foo"</span>, <span class="dv">10</span>,       <span class="dv">8</span>,    <span class="dv">2</span>,    <span class="dv">0</span>,     <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb17-11" data-line-number="11">)</a></code></pre></div>
<p>In this data frame, each row corresponds to a single proteoform. The first column must be labeled <code>name</code> and should contain a short, informative description. All subsequent columns must be labeled by valid PTM names (see below) and contain the number of the respective PTM found in the respective proteoform.</p>
<p>The data frame above describes a range of proteoforms that differ in <em>N</em>-glycosylation, phosphorylation (<code>PHOS</code>) and “fooylation” (<code>foo</code>). Glycosylation is described in terms of the overall monosaccharide composition of the <em>N</em>-glycans; consequently, the data frame includes the columns <code>Hex</code>, <code>HexNAc</code>, and <code>Fuc</code> to indicate the number of hexose, <em>N</em>-acetylhexosamine, and fucose residues, respectively.</p>
<p>Valid PTM names include</p>
<ul>
<li>
<p>monosaccharides as provided in the data set <code>monosaccharides</code>:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1">monosaccharides</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; # A tibble: 83 x 5</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#&gt;    abbreviation type      formula   short_name     systematic_name              </span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;          &lt;chr&gt;                        </span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#&gt;  1 4eLeg        Di-deoxy… C9 H16 N… 4-Epilegionam… 5,7-Diamino-3,5,7,9-tetradeo…</span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="co">#&gt;  2 6dAlt        Deoxyhex… C6 H10 O4 6-Deoxy-L-alt… 6-Deoxy-L-altropyranose      </span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="co">#&gt;  3 6dAltNAc     Deoxyhex… C8 H13 N… N-Acetyl-6-de… 2-Acetamido-2,6-dideoxy-L-al…</span></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="co">#&gt;  4 6dGul        Deoxyhex… C6 H10 O4 6-Deoxy-D-gul… 6-Deoxy-D-gulopyranose       </span></a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="co">#&gt;  5 6dTal        Deoxyhex… C6 H10 O4 6-Deoxy-D-tal… 6-Deoxy-D-talopyranose       </span></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="co">#&gt;  6 6dTalNAc     Deoxyhex… C8 H13 N… N-Acetyl-6-de… 2-Acetamido-2,6-dideoxy-D-ta…</span></a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="co">#&gt;  7 Abe          Di-deoxy… C6 H10 O3 Abequose       3,6-Dideoxy-D-xylo-hexopyran…</span></a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="co">#&gt;  8 Aci          Di-deoxy… C9 H16 N… Acinetaminic … 5,7-Diamino-3,5,7,9-tetradeo…</span></a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="co">#&gt;  9 All          Hexose    C6 H10 O5 D-Allose       D-Allopyranose               </span></a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="co">#&gt; 10 AllA         Hexurona… C5 H8 O6  D-Alluronic a… D-Allopyranuronic acid       </span></a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="co">#&gt; # … with 73 more rows</span></a></code></pre></div>
</li>
<li>
<p>PTMs as provided in the data set <code>ptms</code>:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">ptms</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#&gt; # A tibble: 72 x 3</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt;    abbreviation name                     formula          </span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt;        &lt;chr&gt;                    &lt;chr&gt;            </span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#&gt;  1 ACET         Acetylation              C2 H2 O          </span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#&gt;  2 ADP          ADP-ribosylation         C15 H21 N5 O13 P2</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt;  3 ALLYS        Allysine                 H-3 N-1 O        </span></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">#&gt;  4 AMID         Amidation                H N O-1          </span></a>
<a class="sourceLine" id="cb19-9" data-line-number="9"><span class="co">#&gt;  5 ARCH         S-archaeol               C43 H86 O2       </span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="co">#&gt;  6 BMTH         Beta-methylthiolation    C H2 S           </span></a>
<a class="sourceLine" id="cb19-11" data-line-number="11"><span class="co">#&gt;  7 BIOT         Biotin                   C10 H14 N2 O2 S  </span></a>
<a class="sourceLine" id="cb19-12" data-line-number="12"><span class="co">#&gt;  8 BROM         Bromination              Br H-1           </span></a>
<a class="sourceLine" id="cb19-13" data-line-number="13"><span class="co">#&gt;  9 CETH         N6-1-carboxyethyl lysine C3 H4 O2         </span></a>
<a class="sourceLine" id="cb19-14" data-line-number="14"><span class="co">#&gt; 10 CHOL         Cholesterol              C27 H44          </span></a>
<a class="sourceLine" id="cb19-15" data-line-number="15"><span class="co">#&gt; # … with 62 more rows</span></a></code></pre></div>
</li>
<li>
<p>other PTMs that must be defined in terms of their correction formula:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">my_ptms &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">  <span class="dt">foo =</span> <span class="st">"C42"</span>,  <span class="co"># so this is fooylation!</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3">  <span class="dt">bar =</span> <span class="st">"H-42"</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4">)</a></code></pre></div>
</li>
</ul>
<p>Once all proteoforms have been described in this way, <code><a href="../reference/calculate_proteoform_masses.html">calculate_proteoform_masses()</a></code> calculates their masses, appending a column <code>mass</code> to the data frame of proteoforms:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">pfm_masses &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="st">  </span>proteoforms <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/calculate_proteoform_masses.html">calculate_proteoform_masses</a></span>(mab_formula, <span class="dt">other_ptms =</span> my_ptms)</a>
<a class="sourceLine" id="cb21-4" data-line-number="4"></a>
<a class="sourceLine" id="cb21-5" data-line-number="5">pfm_masses</a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt; # A tibble: 8 x 7</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt;   name          Hex HexNAc   Fuc  PHOS   foo    mass</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="co">#&gt;   &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="co">#&gt; 1 G0F/G0          6      8     1     0     0 147942.</span></a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="co">#&gt; 2 G0F/G0F         6      8     2     0     0 148088.</span></a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="co">#&gt; 3 G0F/G1F         7      8     2     0     0 148250.</span></a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="co">#&gt; 4 G1F/G1F         8      8     2     0     0 148412.</span></a>
<a class="sourceLine" id="cb21-13" data-line-number="13"><span class="co">#&gt; 5 G1F/G2F         9      8     2     0     0 148574.</span></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"><span class="co">#&gt; 6 G2F/G2F        10      8     2     0     0 148737.</span></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="co">#&gt; 7 G2F/G2F+P      10      8     2     1     0 148816.</span></a>
<a class="sourceLine" id="cb21-16" data-line-number="16"><span class="co">#&gt; 8 G2F/G2F+foo    10      8     2     0     1 149241.</span></a></code></pre></div>
<p>Finally, <code><a href="../reference/ionize.html">ionize()</a></code> calculates mass-to-charge ratios of these proteoforms in selected <code>charge_states</code>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">pfm_ions &lt;-</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="st">  </span>pfm_masses <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/ionize.html">ionize</a></span>(<span class="dt">charge_states =</span> 33L<span class="op">:</span>40L)</a>
<a class="sourceLine" id="cb22-4" data-line-number="4"></a>
<a class="sourceLine" id="cb22-5" data-line-number="5">pfm_ions</a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">#&gt; # A tibble: 64 x 11</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="co">#&gt;    name      Hex HexNAc   Fuc  PHOS   foo    mass     z    mz mz_min mz_max</span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">#&gt;    &lt;chr&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">#&gt;  1 G0F/G0      6      8     1     0     0 147942.    33 4484.  4483.  4485.</span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">#&gt;  2 G0F/G0      6      8     1     0     0 147942.    34 4352.  4351.  4353.</span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">#&gt;  3 G0F/G0      6      8     1     0     0 147942.    35 4228.  4227.  4229.</span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">#&gt;  4 G0F/G0      6      8     1     0     0 147942.    36 4111.  4110.  4111.</span></a>
<a class="sourceLine" id="cb22-13" data-line-number="13"><span class="co">#&gt;  5 G0F/G0      6      8     1     0     0 147942.    37 3999.  3999.  4000.</span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14"><span class="co">#&gt;  6 G0F/G0      6      8     1     0     0 147942.    38 3894.  3893.  3895.</span></a>
<a class="sourceLine" id="cb22-15" data-line-number="15"><span class="co">#&gt;  7 G0F/G0      6      8     1     0     0 147942.    39 3794.  3794.  3795.</span></a>
<a class="sourceLine" id="cb22-16" data-line-number="16"><span class="co">#&gt;  8 G0F/G0      6      8     1     0     0 147942.    40 3700.  3699.  3700.</span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17"><span class="co">#&gt;  9 G0F/G0F     6      8     2     0     0 148088.    33 4489.  4488.  4489.</span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18"><span class="co">#&gt; 10 G0F/G0F     6      8     2     0     0 148088.    34 4357.  4356.  4357.</span></a>
<a class="sourceLine" id="cb22-19" data-line-number="19"><span class="co">#&gt; # … with 54 more rows</span></a></code></pre></div>
<p>This function appends columns <code>z</code> (charge state), <code>mz</code> (exact mass-to-charge ratio), <code>mz_min</code>, and <code>mz_max</code>. The latter two columns represent lower and upper tolerances for the mass-to-charge ratio and will be used as integration boundaries (see below). (The width of this tolerance region is governed by the optional argument <code>ppm</code>; see the documentation of <code><a href="../reference/ionize.html">ionize()</a></code> for details.) Since each proteoform typically occurs in several charge states, the resulting data frame will contain more rows than the supplied data frame. (Or, using <a href="https://tidyr.tidyverse.org/articles/tidy-data.html">tidy data</a> terminology: In the input data frame, each observation corresponded to one <em>proteoform</em>; in the output data frame, each observation corresponds to one <em>ion</em>).</p>
</div>
<div id="quantify-proteoforms" class="section level1">
<h1 class="hasAnchor">
<a href="#quantify-proteoforms" class="anchor"></a>Quantify proteoforms</h1>
<p>Proteoform quantification requires mass spectrometric data, which can be loaded by <code><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html">mzR::openMSfile()</a></code>. The exemplary file <code>mab1.mzML</code> contains mass spectra of a monoclonal antibody.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">ms_data &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html">openMSfile</a></span>(</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">  <span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"mzml"</span>, <span class="st">"mab1.mzML"</span>, <span class="dt">package =</span> <span class="st">"fragquaxi"</span>)</a>
<a class="sourceLine" id="cb23-3" data-line-number="3">)</a></code></pre></div>
<p>The figure below shows charge states 37+ to 34+ (from left to right) of a representative scan. Each charge state comprises several discernible peaks, each of which corresponds to a single proteoform (or a set of isobaric proteoforms).</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">sample_spectrum &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/mzR/man/peaks.html">spectra</a></span>(ms_data, <span class="dv">128</span>)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(sample_spectrum) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"mz"</span>, <span class="st">"intensity"</span>)</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">sample_spectrum <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb24-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(mz, intensity)) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb24-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="dv">3950</span>, <span class="dv">4400</span>)</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<p>In order to check whether the previously calculated mass-to-charge ratios match peaks in the mass spectrum, the helper function <code><a href="../reference/plot_ions.html">plot_ions()</a></code> allows you to plot mass spectra overlaid with <span class="math inline">\(m/z\)</span> values and integration boundaries of proteoform ions:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw"><a href="../reference/plot_ions.html">plot_ions</a></span>(</a>
<a class="sourceLine" id="cb25-2" data-line-number="2">  ms_data,</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">  <span class="dt">ions =</span> pfm_ions,</a>
<a class="sourceLine" id="cb25-4" data-line-number="4">  <span class="dt">scans =</span> <span class="dv">126</span><span class="op">:</span><span class="dv">136</span>,      <span class="co"># plot these scans</span></a>
<a class="sourceLine" id="cb25-5" data-line-number="5">  <span class="dt">xlim =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">4100</span>, <span class="dv">4150</span>)  <span class="co"># zoom into charge state 36+</span></a>
<a class="sourceLine" id="cb25-6" data-line-number="6">)</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
<p>Labels on the upper <span class="math inline">\(x\)</span>-axis denote proteoform ions, while solid and dashed red lines indicate mass-to-charge ratios and integration boundaries, respectively</p>
<p><code><a href="../reference/quantify_ions.html">quantify_ions()</a></code> quantifies the previously defined proteoform ions. In the loaded MS data set, only mass spectra that were acquired at retention times between 300 and 350 seconds contain relevant information. Thus, specify these retention time limits <em>via</em> the argument <code>rt_liimits</code>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">abundances &lt;-<span class="st"> </span><span class="kw"><a href="../reference/quantify_ions.html">quantify_ions</a></span>(</a>
<a class="sourceLine" id="cb26-2" data-line-number="2">  ms_data,</a>
<a class="sourceLine" id="cb26-3" data-line-number="3">  pfm_ions,</a>
<a class="sourceLine" id="cb26-4" data-line-number="4">  <span class="dt">rt_limits =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">300</span>, <span class="dv">350</span>)</a>
<a class="sourceLine" id="cb26-5" data-line-number="5">)</a></code></pre></div>
<p>(<code>rt_limits</code> also allows to specify several retention time windows, as described in the documentation.)</p>
<p>Printing the quantification results stored in <code>abundances</code> yields a concise summary:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">abundances</a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="co">#&gt; ℹ Abundances of 64 ions quantified in 352 mass spectra using 1 retention time window.</span></a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="co">#&gt; ── Parameters ──</span></a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-6" data-line-number="6"><span class="co">#&gt; MS data file: /tmp/RtmprYa8wJ/temp_libpath337c325ca21d/fragquaxi/extdata/mzml/mab1.mzML</span></a>
<a class="sourceLine" id="cb27-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-8" data-line-number="8"><span class="co">#&gt; Ions:</span></a>
<a class="sourceLine" id="cb27-9" data-line-number="9"><span class="co">#&gt; # A tibble: 64 x 12</span></a>
<a class="sourceLine" id="cb27-10" data-line-number="10"><span class="co">#&gt;   ion_id name     Hex HexNAc   Fuc  PHOS   foo    mass     z    mz mz_min mz_max</span></a>
<a class="sourceLine" id="cb27-11" data-line-number="11"><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb27-12" data-line-number="12"><span class="co">#&gt; 1 id_1   G0F/G0     6      8     1     0     0 147942.    33 4484.  4483.  4485.</span></a>
<a class="sourceLine" id="cb27-13" data-line-number="13"><span class="co">#&gt; 2 id_2   G0F/G0     6      8     1     0     0 147942.    34 4352.  4351.  4353.</span></a>
<a class="sourceLine" id="cb27-14" data-line-number="14"><span class="co">#&gt; 3 id_3   G0F/G0     6      8     1     0     0 147942.    35 4228.  4227.  4229.</span></a>
<a class="sourceLine" id="cb27-15" data-line-number="15"><span class="co">#&gt; 4 id_4   G0F/G0     6      8     1     0     0 147942.    36 4111.  4110.  4111.</span></a>
<a class="sourceLine" id="cb27-16" data-line-number="16"><span class="co">#&gt; 5 id_5   G0F/G0     6      8     1     0     0 147942.    37 3999.  3999.  4000.</span></a>
<a class="sourceLine" id="cb27-17" data-line-number="17"><span class="co">#&gt; # … with 59 more rows</span></a>
<a class="sourceLine" id="cb27-18" data-line-number="18"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-19" data-line-number="19"><span class="co">#&gt; Retention time limits:</span></a>
<a class="sourceLine" id="cb27-20" data-line-number="20"><span class="co">#&gt; # A tibble: 1 x 4</span></a>
<a class="sourceLine" id="cb27-21" data-line-number="21"><span class="co">#&gt;   rt_min rt_max scan_min scan_max</span></a>
<a class="sourceLine" id="cb27-22" data-line-number="22"><span class="co">#&gt;    &lt;dbl&gt;  &lt;dbl&gt;    &lt;int&gt;    &lt;int&gt;</span></a>
<a class="sourceLine" id="cb27-23" data-line-number="23"><span class="co">#&gt; 1    300    350      117      141</span></a>
<a class="sourceLine" id="cb27-24" data-line-number="24"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-25" data-line-number="25"><span class="co">#&gt; ── Results ──</span></a>
<a class="sourceLine" id="cb27-26" data-line-number="26"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb27-27" data-line-number="27"><span class="co">#&gt; # A tibble: 1 x 66</span></a>
<a class="sourceLine" id="cb27-28" data-line-number="28"><span class="co">#&gt;   rt_min rt_max   id_1   id_2   id_3   id_4   id_5   id_6   id_7   id_8   id_9</span></a>
<a class="sourceLine" id="cb27-29" data-line-number="29"><span class="co">#&gt;    &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb27-30" data-line-number="30"><span class="co">#&gt; 1    300    350 1.54e5 2.05e5 2.57e5 3.34e5 1.84e5 1.77e5 1.21e5 1.14e5 2.90e6</span></a>
<a class="sourceLine" id="cb27-31" data-line-number="31"><span class="co">#&gt; # … with 55 more variables: id_10 &lt;dbl&gt;, id_11 &lt;dbl&gt;, id_12 &lt;dbl&gt;, id_13 &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb27-32" data-line-number="32"><span class="co">#&gt; #   id_14 &lt;dbl&gt;, …</span></a></code></pre></div>
<p>For data analysis, it may be convenient to convert these results to a data frame:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>(abundances)</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="co">#&gt; # A tibble: 64 x 13</span></a>
<a class="sourceLine" id="cb28-3" data-line-number="3"><span class="co">#&gt;    name    Hex HexNAc   Fuc  PHOS   foo   mass     z    mz mz_min mz_max</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#&gt;    &lt;chr&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="co">#&gt;  1 G0F/…     6      8     1     0     0 1.48e5    33 4484.  4483.  4485.</span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="co">#&gt;  2 G0F/…     6      8     1     0     0 1.48e5    34 4352.  4351.  4353.</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="co">#&gt;  3 G0F/…     6      8     1     0     0 1.48e5    35 4228.  4227.  4229.</span></a>
<a class="sourceLine" id="cb28-8" data-line-number="8"><span class="co">#&gt;  4 G0F/…     6      8     1     0     0 1.48e5    36 4111.  4110.  4111.</span></a>
<a class="sourceLine" id="cb28-9" data-line-number="9"><span class="co">#&gt;  5 G0F/…     6      8     1     0     0 1.48e5    37 3999.  3999.  4000.</span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="co">#&gt;  6 G0F/…     6      8     1     0     0 1.48e5    38 3894.  3893.  3895.</span></a>
<a class="sourceLine" id="cb28-11" data-line-number="11"><span class="co">#&gt;  7 G0F/…     6      8     1     0     0 1.48e5    39 3794.  3794.  3795.</span></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="co">#&gt;  8 G0F/…     6      8     1     0     0 1.48e5    40 3700.  3699.  3700.</span></a>
<a class="sourceLine" id="cb28-13" data-line-number="13"><span class="co">#&gt;  9 G0F/…     6      8     2     0     0 1.48e5    33 4489.  4488.  4489.</span></a>
<a class="sourceLine" id="cb28-14" data-line-number="14"><span class="co">#&gt; 10 G0F/…     6      8     2     0     0 1.48e5    34 4357.  4356.  4357.</span></a>
<a class="sourceLine" id="cb28-15" data-line-number="15"><span class="co">#&gt; # … with 54 more rows, and 2 more variables: abundance_data &lt;list&gt;,</span></a>
<a class="sourceLine" id="cb28-16" data-line-number="16"><span class="co">#&gt; #   xic_data &lt;list&gt;</span></a></code></pre></div>
<p>This conversion returns the data frame describing proteoform ions, plus two additional nested columns <code>abundance_data</code> and <code>xic_data</code>.</p>
<ul>
<li>
<p>The column <code>abundance_data</code> stores data on ion abundances. Each value is a tibble with one row per retention time window (given by the values in columns <code>rt_min</code> and <code>rt_max</code>); column <code>abundance</code> contains corresponding XIC area.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">abundances <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb29-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(name, z, abundance_data) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span>(abundance_data)</a>
<a class="sourceLine" id="cb29-5" data-line-number="5"><span class="co">#&gt; # A tibble: 64 x 7</span></a>
<a class="sourceLine" id="cb29-6" data-line-number="6"><span class="co">#&gt;    name        z rt_min rt_max scan_min scan_max abundance</span></a>
<a class="sourceLine" id="cb29-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;   &lt;int&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;int&gt;    &lt;int&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb29-8" data-line-number="8"><span class="co">#&gt;  1 G0F/G0     33    300    350      117      141   153744.</span></a>
<a class="sourceLine" id="cb29-9" data-line-number="9"><span class="co">#&gt;  2 G0F/G0     34    300    350      117      141   204512.</span></a>
<a class="sourceLine" id="cb29-10" data-line-number="10"><span class="co">#&gt;  3 G0F/G0     35    300    350      117      141   256930.</span></a>
<a class="sourceLine" id="cb29-11" data-line-number="11"><span class="co">#&gt;  4 G0F/G0     36    300    350      117      141   334421.</span></a>
<a class="sourceLine" id="cb29-12" data-line-number="12"><span class="co">#&gt;  5 G0F/G0     37    300    350      117      141   183758.</span></a>
<a class="sourceLine" id="cb29-13" data-line-number="13"><span class="co">#&gt;  6 G0F/G0     38    300    350      117      141   177475.</span></a>
<a class="sourceLine" id="cb29-14" data-line-number="14"><span class="co">#&gt;  7 G0F/G0     39    300    350      117      141   121394.</span></a>
<a class="sourceLine" id="cb29-15" data-line-number="15"><span class="co">#&gt;  8 G0F/G0     40    300    350      117      141   114270.</span></a>
<a class="sourceLine" id="cb29-16" data-line-number="16"><span class="co">#&gt;  9 G0F/G0F    33    300    350      117      141  2903791.</span></a>
<a class="sourceLine" id="cb29-17" data-line-number="17"><span class="co">#&gt; 10 G0F/G0F    34    300    350      117      141  4762999.</span></a>
<a class="sourceLine" id="cb29-18" data-line-number="18"><span class="co">#&gt; # … with 54 more rows</span></a></code></pre></div>
<p>Use this data to visualize proteoform abundances:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">abundances <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb30-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span>(abundance_data) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb30-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb30-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="dt">abundance =</span> <span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(abundance)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb30-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">frac_ab =</span> abundance <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(abundance) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb30-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(name, frac_ab)) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb30-9" data-line-number="9"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">""</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-10" data-line-number="10"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"fractional abundance (%)"</span>)</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
</li>
<li>
<p>The column <code>xic_data</code> stores retention times (column <code>rt</code>) and intensities (column <code>int</code>) of the XIC associated with the respective ion:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">abundances <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(name, z, xic_data) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb31-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span>(xic_data)</a>
<a class="sourceLine" id="cb31-5" data-line-number="5"><span class="co">#&gt; # A tibble: 22,528 x 5</span></a>
<a class="sourceLine" id="cb31-6" data-line-number="6"><span class="co">#&gt;    name       z  scan    rt   int</span></a>
<a class="sourceLine" id="cb31-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;  &lt;int&gt; &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb31-8" data-line-number="8"><span class="co">#&gt;  1 G0F/G0    33     1  2.58     0</span></a>
<a class="sourceLine" id="cb31-9" data-line-number="9"><span class="co">#&gt;  2 G0F/G0    33     2  5.20     0</span></a>
<a class="sourceLine" id="cb31-10" data-line-number="10"><span class="co">#&gt;  3 G0F/G0    33     3  7.81     0</span></a>
<a class="sourceLine" id="cb31-11" data-line-number="11"><span class="co">#&gt;  4 G0F/G0    33     4 10.4      0</span></a>
<a class="sourceLine" id="cb31-12" data-line-number="12"><span class="co">#&gt;  5 G0F/G0    33     5 13.0      0</span></a>
<a class="sourceLine" id="cb31-13" data-line-number="13"><span class="co">#&gt;  6 G0F/G0    33     6 15.6      0</span></a>
<a class="sourceLine" id="cb31-14" data-line-number="14"><span class="co">#&gt;  7 G0F/G0    33     7 18.3      0</span></a>
<a class="sourceLine" id="cb31-15" data-line-number="15"><span class="co">#&gt;  8 G0F/G0    33     8 20.9      0</span></a>
<a class="sourceLine" id="cb31-16" data-line-number="16"><span class="co">#&gt;  9 G0F/G0    33     9 23.5      0</span></a>
<a class="sourceLine" id="cb31-17" data-line-number="17"><span class="co">#&gt; 10 G0F/G0    33    10 26.1      0</span></a>
<a class="sourceLine" id="cb31-18" data-line-number="18"><span class="co">#&gt; # … with 22,518 more rows</span></a></code></pre></div>
<p>Use this data to visualize all XICs:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">abundances <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb32-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span>(xic_data) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb32-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(rt, int)) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">color =</span> <span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(z))) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(<span class="kw"><a href="https://dplyr.tidyverse.org/reference/vars.html">vars</a></span>(name)) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"retention time (s)"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"intensity (AU)"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-9" data-line-number="9"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="dv">300</span>, <span class="dv">350</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-10" data-line-number="10"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-32-1.png" width="700"></p>
</li>
</ul>
</div>
<div id="working-with-several-ms-files" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-several-ms-files" class="anchor"></a>Working with several MS files</h1>
<p>Ions may also be quantified in a batch of MS files, and the resulting abundances may be used to calculate summary statistics.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">ms_files &lt;-</a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"mzml"</span>, <span class="dt">package =</span> <span class="st">"fragquaxi"</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb33-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(<span class="dt">pattern =</span> <span class="st">"mab</span><span class="ch">\\</span><span class="st">d"</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb33-4" data-line-number="4"></a>
<a class="sourceLine" id="cb33-5" data-line-number="5">all_abundances &lt;-</a>
<a class="sourceLine" id="cb33-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://purrr.tidyverse.org/reference/map.html">map_dfr</a></span>(</a>
<a class="sourceLine" id="cb33-7" data-line-number="7">    ms_files,</a>
<a class="sourceLine" id="cb33-8" data-line-number="8">    <span class="cf">function</span>(file) {</a>
<a class="sourceLine" id="cb33-9" data-line-number="9">      ms_file &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html">openMSfile</a></span>(file)</a>
<a class="sourceLine" id="cb33-10" data-line-number="10">      xic &lt;-<span class="st"> </span><span class="kw"><a href="../reference/quantify_ions.html">quantify_ions</a></span>(ms_file, pfm_ions, <span class="dt">rt_limits =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">300</span>, <span class="dv">350</span>))</a>
<a class="sourceLine" id="cb33-11" data-line-number="11">      xic <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-12" data-line-number="12"><span class="st">        </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-13" data-line-number="13"><span class="st">        </span><span class="kw"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span>(abundance_data) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-14" data-line-number="14"><span class="st">        </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-15" data-line-number="15"><span class="st">        </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="dt">abundance =</span> <span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(abundance)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb33-16" data-line-number="16"><span class="st">        </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">abundance =</span> abundance <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(abundance) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>)</a>
<a class="sourceLine" id="cb33-17" data-line-number="17">    },</a>
<a class="sourceLine" id="cb33-18" data-line-number="18">    <span class="dt">.id =</span> <span class="st">"sample"</span></a>
<a class="sourceLine" id="cb33-19" data-line-number="19">  )</a>
<a class="sourceLine" id="cb33-20" data-line-number="20"></a>
<a class="sourceLine" id="cb33-21" data-line-number="21">all_abundances</a>
<a class="sourceLine" id="cb33-22" data-line-number="22"><span class="co">#&gt; # A tibble: 24 x 3</span></a>
<a class="sourceLine" id="cb33-23" data-line-number="23"><span class="co">#&gt;    sample name        abundance</span></a>
<a class="sourceLine" id="cb33-24" data-line-number="24"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;           &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb33-25" data-line-number="25"><span class="co">#&gt;  1 1      G0F/G0          2.56 </span></a>
<a class="sourceLine" id="cb33-26" data-line-number="26"><span class="co">#&gt;  2 1      G0F/G0F        57.8  </span></a>
<a class="sourceLine" id="cb33-27" data-line-number="27"><span class="co">#&gt;  3 1      G0F/G1F        23.4  </span></a>
<a class="sourceLine" id="cb33-28" data-line-number="28"><span class="co">#&gt;  4 1      G1F/G1F        10.5  </span></a>
<a class="sourceLine" id="cb33-29" data-line-number="29"><span class="co">#&gt;  5 1      G1F/G2F         4.07 </span></a>
<a class="sourceLine" id="cb33-30" data-line-number="30"><span class="co">#&gt;  6 1      G2F/G2F         1.41 </span></a>
<a class="sourceLine" id="cb33-31" data-line-number="31"><span class="co">#&gt;  7 1      G2F/G2F+foo     0.100</span></a>
<a class="sourceLine" id="cb33-32" data-line-number="32"><span class="co">#&gt;  8 1      G2F/G2F+P       0.115</span></a>
<a class="sourceLine" id="cb33-33" data-line-number="33"><span class="co">#&gt;  9 2      G0F/G0          2.00 </span></a>
<a class="sourceLine" id="cb33-34" data-line-number="34"><span class="co">#&gt; 10 2      G0F/G0F        57.6  </span></a>
<a class="sourceLine" id="cb33-35" data-line-number="35"><span class="co">#&gt; # … with 14 more rows</span></a></code></pre></div>
<p>The following bar chart displays ion abundances in each sample:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">all_abundances <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(sample, abundance)) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">fill =</span> name), <span class="dt">position =</span> <span class="st">"dodge"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"fractional abundance (%)"</span>)</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-34-1.png" width="700"></p>
<p>Since the three MS data files correspond to replicates, calculate mean proteoform abundances and their standard deviations:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">all_abundances <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(name) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="dt">mean =</span> <span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(abundance), <span class="dt">sd =</span> <span class="kw"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span>(abundance)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(name, mean)) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">fill =</span> name), <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_linerange.html">geom_errorbar</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">ymin =</span> mean <span class="op">-</span><span class="st"> </span>sd, <span class="dt">ymax =</span> mean <span class="op">+</span><span class="st"> </span>sd), <span class="dt">width =</span> <span class="fl">.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"fractional abundance (%)"</span>)</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-35-1.png" width="700"></p>
</div>
<div id="direct-infusion" class="section level1">
<h1 class="hasAnchor">
<a href="#direct-infusion" class="anchor"></a>Direct infusion</h1>
<p>In certain experiments (e.g., direct infusion with rolling averaging), it may be sensible to quantify proteoforms by integrating a single mass spectrum instead of an XIC. <code>fragquaxi</code> also facilitates this approach, since each intensity value in the XIC data reported by <code>quantify_ions(...) %&gt;% as_table()</code> corresponds to the abundance of an ion in a single spectrum.</p>
<p>Below, we load a series of 105 spectra from a direct infusion measurement of NISTmAb and define proteoforms to be quantified in charge states 24+ to 30+:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1">di_data &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html">openMSfile</a></span>(</a>
<a class="sourceLine" id="cb36-2" data-line-number="2">  <span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"mzml"</span>, <span class="st">"mab_di.mzML"</span>, <span class="dt">package =</span> <span class="st">"fragquaxi"</span>)</a>
<a class="sourceLine" id="cb36-3" data-line-number="3">)</a>
<a class="sourceLine" id="cb36-4" data-line-number="4"></a>
<a class="sourceLine" id="cb36-5" data-line-number="5">di_proteoforms &lt;-<span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">tribble</a></span>(</a>
<a class="sourceLine" id="cb36-6" data-line-number="6">  <span class="op">~</span>Hex, <span class="op">~</span>HexNAc, <span class="op">~</span>Fuc, <span class="op">~</span>name,</a>
<a class="sourceLine" id="cb36-7" data-line-number="7">   <span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="st">"A2G0F/A2G0F"</span>,</a>
<a class="sourceLine" id="cb36-8" data-line-number="8">   <span class="dv">7</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="st">"A2G0F/A2G1F"</span>,</a>
<a class="sourceLine" id="cb36-9" data-line-number="9">   <span class="dv">8</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="st">"A2G1F/A2G1F"</span>,</a>
<a class="sourceLine" id="cb36-10" data-line-number="10">   <span class="dv">9</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="st">"A2G1F/A2G2F"</span>,</a>
<a class="sourceLine" id="cb36-11" data-line-number="11">  <span class="dv">10</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="st">"A2G2F/A2G2F"</span>,</a>
<a class="sourceLine" id="cb36-12" data-line-number="12">  <span class="dv">11</span>, <span class="dv">8</span>, <span class="dv">2</span>, <span class="st">"A2G2F/A2G2F+1Hex"</span>,</a>
<a class="sourceLine" id="cb36-13" data-line-number="13">   <span class="dv">6</span>, <span class="dv">7</span>, <span class="dv">2</span>, <span class="st">"A1G0F/A2G0F"</span></a>
<a class="sourceLine" id="cb36-14" data-line-number="14">) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-15" data-line-number="15"><span class="st">  </span><span class="kw"><a href="../reference/calculate_proteoform_masses.html">calculate_proteoform_masses</a></span>(<span class="st">"C6464 H9984 N1700 O2006 S48"</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-16" data-line-number="16"><span class="st">  </span><span class="kw"><a href="../reference/ionize.html">ionize</a></span>(24L<span class="op">:</span>30L)</a></code></pre></div>
<p>Since these spectra were obtained by rolling averaging, it is sufficient to quantify proteoforms in the last recorded spectrum, where most of the noise has been eliminated.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="kw"><a href="../reference/plot_ions.html">plot_ions</a></span>(di_data, di_proteoforms, <span class="dt">scans =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="dv">105</span>), <span class="dt">xlim =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">5450</span>, <span class="dv">5550</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(</a>
<a class="sourceLine" id="cb37-3" data-line-number="3">    <span class="kw"><a href="https://dplyr.tidyverse.org/reference/vars.html">vars</a></span>(scan),</a>
<a class="sourceLine" id="cb37-4" data-line-number="4">    <span class="dt">ncol =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb37-5" data-line-number="5">    <span class="dt">strip.position =</span> <span class="st">"right"</span>,</a>
<a class="sourceLine" id="cb37-6" data-line-number="6">    <span class="dt">scales =</span> <span class="st">"free_y"</span></a>
<a class="sourceLine" id="cb37-7" data-line-number="7">  )</a></code></pre></div>
<p><img src="fragquaxi_files/figure-html/unnamed-chunk-37-1.png" width="700"></p>
<p>(Note how the plot returned by <code><a href="../reference/plot_ions.html">plot_ions()</a></code> may be further customized <em>via</em> the usual <code>ggplot2</code> syntax – here, a facet specification is included.)</p>
<p>Following quantification, obtain XIC data and filter for the last scan number:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1">di_abundances &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb38-2" data-line-number="2"><span class="st">  </span>di_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="../reference/quantify_ions.html">quantify_ions</a></span>(di_proteoforms) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>(<span class="dt">abundance_col =</span> <span class="ot">NULL</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://tidyr.tidyverse.org/reference/nest.html">unnest</a></span>(xic_data) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb38-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(scan <span class="op">==</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(scan))</a>
<a class="sourceLine" id="cb38-7" data-line-number="7"></a>
<a class="sourceLine" id="cb38-8" data-line-number="8">di_abundances</a>
<a class="sourceLine" id="cb38-9" data-line-number="9"><span class="co">#&gt; # A tibble: 49 x 12</span></a>
<a class="sourceLine" id="cb38-10" data-line-number="10"><span class="co">#&gt;      Hex HexNAc   Fuc name     mass     z    mz mz_min mz_max  scan    rt    int</span></a>
<a class="sourceLine" id="cb38-11" data-line-number="11"><span class="co">#&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb38-12" data-line-number="12"><span class="co">#&gt;  1     6      7     2 A1G0F… 1.48e5    24 6161.  6160.  6162.   105  293.   281.</span></a>
<a class="sourceLine" id="cb38-13" data-line-number="13"><span class="co">#&gt;  2     6      7     2 A1G0F… 1.48e5    25 5914.  5913.  5916.   105  293.  1571.</span></a>
<a class="sourceLine" id="cb38-14" data-line-number="14"><span class="co">#&gt;  3     6      7     2 A1G0F… 1.48e5    26 5687.  5686.  5688.   105  293.  2304.</span></a>
<a class="sourceLine" id="cb38-15" data-line-number="15"><span class="co">#&gt;  4     6      7     2 A1G0F… 1.48e5    27 5476.  5475.  5477.   105  293.  2043.</span></a>
<a class="sourceLine" id="cb38-16" data-line-number="16"><span class="co">#&gt;  5     6      7     2 A1G0F… 1.48e5    28 5281.  5280.  5282.   105  293.  1205.</span></a>
<a class="sourceLine" id="cb38-17" data-line-number="17"><span class="co">#&gt;  6     6      7     2 A1G0F… 1.48e5    29 5099.  5098.  5100.   105  293.   621.</span></a>
<a class="sourceLine" id="cb38-18" data-line-number="18"><span class="co">#&gt;  7     6      7     2 A1G0F… 1.48e5    30 4929.  4928.  4930.   105  293.     0 </span></a>
<a class="sourceLine" id="cb38-19" data-line-number="19"><span class="co">#&gt;  8     6      8     2 A2G0F… 1.48e5    24 6169.  6168.  6171.   105  293.  3684.</span></a>
<a class="sourceLine" id="cb38-20" data-line-number="20"><span class="co">#&gt;  9     6      8     2 A2G0F… 1.48e5    25 5923.  5921.  5924.   105  293. 16880.</span></a>
<a class="sourceLine" id="cb38-21" data-line-number="21"><span class="co">#&gt; 10     6      8     2 A2G0F… 1.48e5    26 5695.  5694.  5696.   105  293. 27466.</span></a>
<a class="sourceLine" id="cb38-22" data-line-number="22"><span class="co">#&gt; # … with 39 more rows</span></a></code></pre></div>
<p>The values in column <code>int</code> represent proteoform ion abundances in the last scan.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">di_abundances <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(name) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="dt">abundance =</span> <span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(int)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb39-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">abundance =</span> abundance <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(abundance) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>)</a>
<a class="sourceLine" id="cb39-5" data-line-number="5"><span class="co">#&gt; # A tibble: 7 x 2</span></a>
<a class="sourceLine" id="cb39-6" data-line-number="6"><span class="co">#&gt;   name             abundance</span></a>
<a class="sourceLine" id="cb39-7" data-line-number="7"><span class="co">#&gt;   &lt;chr&gt;                &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb39-8" data-line-number="8"><span class="co">#&gt; 1 A1G0F/A2G0F           1.67</span></a>
<a class="sourceLine" id="cb39-9" data-line-number="9"><span class="co">#&gt; 2 A2G0F/A2G0F          19.1 </span></a>
<a class="sourceLine" id="cb39-10" data-line-number="10"><span class="co">#&gt; 3 A2G0F/A2G1F          32.5 </span></a>
<a class="sourceLine" id="cb39-11" data-line-number="11"><span class="co">#&gt; 4 A2G1F/A2G1F          26.9 </span></a>
<a class="sourceLine" id="cb39-12" data-line-number="12"><span class="co">#&gt; 5 A2G1F/A2G2F          12.9 </span></a>
<a class="sourceLine" id="cb39-13" data-line-number="13"><span class="co">#&gt; 6 A2G2F/A2G2F           4.98</span></a>
<a class="sourceLine" id="cb39-14" data-line-number="14"><span class="co">#&gt; 7 A2G2F/A2G2F+1Hex      1.95</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#molecular-formulas">Molecular formulas</a></li>
      <li><a href="#define-proteoforms-and-ions">Define proteoforms and ions</a></li>
      <li><a href="#quantify-proteoforms">Quantify proteoforms</a></li>
      <li><a href="#working-with-several-ms-files">Working with several MS files</a></li>
      <li><a href="#direct-infusion">Direct infusion</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://wolfgang.esser-skala.at">Wolfgang Esser-Skala</a>, <a href="https://www.uni-salzburg.at/"><img src="https://www.uni-salzburg.at/fileadmin/templates/Adhurricane_templates/images/logo.gif" alt="PLUS" height="24"></a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
